src/tpm2tss-genkey.c

```diff
@@ -390,6 +396,10 @@ main(int argc, char **argv)
         VERB("Generating the ecdsa key\n");
         tpm2Data = genkey_ecdsa();
         break;
+    case TPM2_ALG_SM2:
+        VERB("Generating the sm2 key\n");
+        tpm2Data = genkey_ecdsa();
+        break;
```

Comment:
Here you call the same function in the two cases, it looks wrong. Are the key algorithm passed through a global variable?